identity/client/src/pages/groups/modals/EditModal.tsx

```diff
@@ -29,14 +29,15 @@ const EditModal: FC<UseEntityModalProps<Group>> = ({
   });
 
   const [groupName, setGroupName] = useState(group.name);
-  const [groupId, setGroupId] = useState(group.groupKey);
+  const [groupId, setGroupId] = useState(group.groupId);
   const [description, setDescription] = useState(group.description);
   const [isGroupIdValid, setIsGroupIdValid] = useState(true);
 
   const handleSubmit = async () => {
     const { success } = await callUpdateGroup({
-      groupKey: group.groupKey,
+      groupId: group.groupId,
```

Comment:
`.trim()` is missing here, it is used in the add modal [here](https://github.com/camunda/camunda/pull/32087/files#diff-b0ce54845aab8763437fa503f720dca0b38bc152bf90a399b422f3657a1fff89R32).

Looking at all the trims I would suggest moving "trimming" to the api call itself, to make the usage across different components safe (in case someone forgets to trim a value that should be trimmed)